File: src/main/java/ru/org/linux/search/ElasticSearchConfiguration.java
package ru.org.linux.search;

import org.elasticsearch.client.Client;
import org.elasticsearch.client.transport.TransportClient;
import org.elasticsearch.common.transport.InetSocketTransportAddress;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class ElasticSearchConfiguration {
  @Bean(destroyMethod = "close")
  public Client elastic() {
    return new TransportClient()
            .addTransportAddress(new InetSocketTransportAddress("127.0.0.1", 9300));
  }
}

File: src/main/scala/ru/org/linux/search/ElasticsearchConfiguration.scala
/*
 * Copyright 1998-2024 Linux.org.ru
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package ru.org.linux.search

import com.sksamuel.elastic4s.http.JavaClient
import com.sksamuel.elastic4s.{ElasticClient, ElasticProperties}
import org.apache.http.client.config.RequestConfig
import org.elasticsearch.client.RestClientBuilder.RequestConfigCallback
import org.springframework.context.annotation.{Bean, Configuration}
import ru.org.linux.spring.SiteConfig

@Configuration
class ElasticsearchConfiguration(config: SiteConfig) {
  @Bean(destroyMethod = "close")
  def client: ElasticClient = {
    ElasticClient(JavaClient(ElasticProperties(config.getElasticsearch), new RequestConfigCallback() {
      override def customizeRequestConfig(requestConfigBuilder: RequestConfig.Builder): RequestConfig.Builder = {
        requestConfigBuilder.setSocketTimeout(120000)
      }
    }))
  }
}
